zebra-state/src/service/finalized_state/tests/vectors.rs

```diff
@@ -4,20 +4,48 @@
 //! We don't need to check empty trees, because the database format snapshot tests
 //! use empty trees.
 
-use halo2::pasta::{group::ff::PrimeField, pallas};
 use hex::FromHex;
 use rand::random;
 
-use zebra_chain::{orchard, sapling, sprout};
+use halo2::pasta::{group::ff::PrimeField, pallas};
+
+use zebra_chain::{
+    orchard::tree::NoteCommitmentTree as OrchardNoteCommitmentTree,
+    sapling::tree::NoteCommitmentTree as SaplingNoteCommitmentTree,
+    sprout::{
+        tree::NoteCommitmentTree as SproutNoteCommitmentTree,
+        NoteCommitment as SproutNoteCommitment,
+    },
+};
 
 use crate::service::finalized_state::disk_format::{FromDisk, IntoDisk};
 
+// Currently, these tests check these structs are equal:
+//  * commitments -> tree struct
+//  * commitments -> tree struct -> seralize -> deserialize -> tree struct
+// And these serialized formats are equal:
+//  * fixed serialized test vector
+//  * commitments -> tree struct -> seralize
+//  * commitments -> tree struct -> seralize -> deserialize -> tree struct -> serialize
+//
+// TODO: apply these tests to the new tree structs, and update the serialization format
+//       (keeping the tests for the old format is optional, because the tests below cover it)
+//
+// TODO: test that old and new serializations produce the same format:
+// Tree roots built from the same commitments should match:
+//   * commitments -> old tree struct -> new tree struct -> un-cached root
+//   * commitments -> new tree struct -> un-cached root
+// Even when serialized and deserialized:
+//   * commitments -> old tree struct -> old serialize -> old deserialize -> old tree struct ->  new tree struct -> un-cached root
+//   * commitments -> new tree struct -> new serialize -> new deserialize -> new tree struct -> un-cached root
+//   * commitments -> new tree struct -> un-cached root
```

Comment:
@oxarbitrage I think these are the tests we need to make sure the format conversions work correctly, and the old and new trees produce the same roots.

Did you want to have a go at writing these tests? I'm happy to help.

The tests need the old and new tree types, the old to new conversion, and old and new serialisations and deserializations. If we don't want to implement the old serialization, we can re-use the old test vectors instead.